### DISCUSSION TITLE
Error message after re-install of Zwave JS UI

### DISCUSSION THREAD

**Discussion by nesheimt** [zwave-js/zwave-js-ui#3999] [Q&A] [answered]
I had a working installation of Zwave JS UI on my Win11 computer. For some reason I deleted the whole directory and did a new clone from Github. I then did the npm install and npm run build. 

But this time the npm run build was finished by an error message: _**Node.js v22.11.0
ERROR: "build:ui" exited with 1**_ and Zwave JS UI does not start properly showing this:
![image](https://github.com/user-attachments/assets/3f830b45-b3dc-41ed-bd50-74b7634ede58)

And when entering the URL to the UI these messages appear>
E**rror: Not Found
    at C:\zwave\zwave-js-ui\server\app.js:1238:17
    at Layer.handle [as handle_request] (C:\zwave\zwave-js-ui\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (C:\zwave\zwave-js-ui\node_modules\express\lib\router\index.js:328:13)
    at C:\zwave\zwave-js-ui\node_modules\express\lib\router\index.js:286:9
    at Function.process_params (C:\zwave\zwave-js-ui\node_modules\express\lib\router\index.js:346:12)
    at next (C:\zwave\zwave-js-ui\node_modules\express\lib\router\index.js:280:10)
    at session (C:\zwave\zwave-js-ui\node_modules\express-session\index.js:487:7)
    at Layer.handle [as handle_request] (C:\zwave\zwave-js-ui\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (C:\zwave\zwave-js-ui\node_modules\express\lib\router\index.js:328:13)
    at C:\zwave\zwave-js-ui\node_modules\express\lib\router\index.js:286:9**

I have tried some suggestions found on the web, but no luck. Please advice.

**Comment by kpine** [comment, marked as answer]
Perhaps you should try the latest, 9.27.1. 

Why are you building it manually instead of just downloading the binary, or using `npm install -g zwave-js-ui`?

**Reply by nesheimt** [reply]
Need to tweak on the compat config for a device.

**Reply by kpine** [reply]
You don't need to install from source to do that. You can drop the config file into the device config directory.

**Reply by nesheimt** [reply]
Thanks for the advices....this is a completely "new world" to me. Will try it out!

**Reply by nesheimt** [reply]
@kpine Just to be sure...if I use the binary file for Windows I can just take the config file, in this case _tf016_tf021.json_, and drop it here:
![image](https://github.com/user-attachments/assets/3c649de4-53b3-4458-9832-7a3ac6db866d)

**Reply by kpine** [reply]
Yes. Then restart ZUI. Check the driver logs to make sure the "User-provided" file was loaded for the node.

```text
CNTRLR   [Node 002] Embedded device config loaded
CNTRLR   [Node 009] User-provided device config loaded
```

**Comment by robertsLando** [comment]
9.27.0 was broken, use 9.27.1

**Reply by nesheimt** [reply]
It doesn't work for me:

C:\zwave\zwave-js-ui>npm run build

> zwave-js-ui@9.27.1 build
> npm-run-all build:*


> zwave-js-ui@9.27.1 build:server
> tsc

api/lib/ZwaveClient.ts:21:2 - error TS2305: Module '"@zwave-js/core"' has no exported member 'tryUnzipFirmwareFile'.

21  tryUnzipFirmwareFile,
    ~~~~~~~~~~~~~~~~~~~~

api/lib/ZwaveClient.ts:3852:55 - error TS2345: Argument of type 'Uint8Array | Buffer' is not assignable to parameter of type 'Buffer'.
  Type 'Uint8Array' is missing the following properties from type 'Buffer': write, toJSON, equals, compare, and 66 more.

3852     const format = guessFirmwareFileFormat(file.name, file.data)
                                                           ~~~~~~~~~

api/lib/ZwaveClient.ts:3853:32 - error TS2345: Argument of type 'Uint8Array | Buffer' is not assignable to parameter of type 'Buffer'.
  Type 'Uint8Array' is not assignable to type 'Buffer'.

3853     firmware = extractFirmware(file.data, format)
                                    ~~~~~~~~~

api/lib/ZwaveClient.ts:3916:40 - error TS2345: Argument of type 'Uint8Array | Buffer' is not assignable to parameter of type 'Buffer'.
  Type 'Uint8Array' is not assignable to type 'Buffer'.

3916       const firmware = extractFirmware(data, format)
                                            ~~~~

api/lib/ZwaveClient.ts:5642:30 - error TS2339: Property 'tryGetNode' does not exist on type 'Endpoint'.

5642   const zwaveNode = endpoint.tryGetNode()
                                  ~~~~~~~~~~


Found 5 errors in the same file, starting at: api/lib/ZwaveClient.ts:21

ERROR: "build:server" exited with 2.

**Reply by robertsLando** [reply]
npm install?

**Reply by nesheimt** [reply]
Yes, thats right. (has been working fine till now)

**Comment by nesheimt** [comment]
Thanks...I thought it was me doing something wrong, but yes, seems to something with the 27 release. 
I rolled back to 26 and that worked fine.

**Reply by robertsLando** [reply]
9.27.1 is already out use that :)
